tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/longlivedtoken/LongLivedTokenStore.scala

```diff
@@ -0,0 +1,111 @@
+package com.linagora.tmail.james.jmap.longlivedtoken
+
+import com.google.common.base.Preconditions
+import org.apache.commons.lang3.tuple.ImmutableTriple
```

Comment:
I bet scala have native triple. Also we could have case classes to cover the usage of this... 

(Maybe using case classes is better)

Reply:
yes, I see  Tuple3, But I think with `ImmutableTriple` it is like as "readOnly"

Reply:
In scala you can write `(a, b, c)`

But I prefer case classes